src/deserialize/parse.rs

```diff
@@ -55,9 +56,36 @@ fn read_key(chars: &mut std::iter::Enumerate<std::str::Chars>, c_len: usize) ->
 
 fn read_str(chars: &mut std::iter::Enumerate<std::str::Chars>) -> Edn {
     let string = chars
-        .take_while(|c| c.1 != '\"')
-        .map(|c| c.1)
-        .collect::<String>();
+        .fold_while((false, String::new()), |(last_was_escape, mut s), c| {
```

Comment:
I would prefere `c` to be more expressive, so I would use `(_, c)`. Then all `c.1`s would become just `c`